package.json

```diff
@@ -41,7 +41,10 @@
     "generate-component-lint": "cd component-generator && npm run lint",
     "generate-changelog": "node generate-changelog.js",
     "i18n_compile": "formatjs compile-folder --format transifex ./src/i18n/messages ./src/i18n/messages",
-    "i18n_extract": "formatjs extract 'src/**/*.{jsx,js}' --out-file ./src/i18n/transifex_input.json --format transifex"
+    "i18n_extract": "formatjs extract 'src/**/*.{jsx,js,tsx,ts}' --out-file ./src/i18n/transifex_input.json --format transifex",
+    "type-check": "tsc --noEmit",
+    "type-check:watch": "npm run type-check -- --watch",
+    "build-types": "tsc --emitDeclarationOnly"
```

Comment:
It might be a good idea to add documentation about what these commands do and when/why we should use them somewhere in the repo. It very well could be the starting documentation for how to work with TypeScript for all of Open edX.

Related, did we write an ADR (Architectural Decision Record) about TypeScript? If not, that's probably something we should do. Is this something you might want to take on? Let's treat that as a task separate from this PR, though 😄